metrics.py

```diff
@@ -0,0 +1,102 @@
+"""Collection of metrics for evaluating the performance of probabilistic models."""
+
+from arviz_base import convert_to_datatree, dataset_to_dataframe, extract, rcParams
+from scipy.stats import wasserstein_distance, wasserstein_distance_nd
+
+
+def wasserstein(
+    data_1,
+    data_2,
+    group="posterior",
+    var_names=None,
+    sample_dims=None,
+    joint=True,
+    num_samples=500,
+    random_seed=212480,
+):
+    """
+    Compute the Wasserstein-1 distance.
+
+    Parameters
+    ----------
+    data : Dataset, DataTree, or InferenceData
+    group : hashable, default "posterior"
+        Group on which to compute the ESS.
+    var_names : str or list of str, optional
+        Names of the variables for which the ess should be computed.
+    sample_dims : iterable of hashable, optional
+        Dimensions to be considered sample dimensions and are to be reduced.
+        Default ``rcParams["data.sample_dims"]``.
+    joint : bool, default True
+        If True, compute the joint Wasserstein distance. If False, compute the
+        Wasserstein distance for each marginal. The computation is faster when `joint=False`
+        but it is equivalent to assume that the marginals are independent, which usually
+        is not the case.
```

Comment:
I think I would move the last sentence to the [notes](https://numpydoc.readthedocs.io/en/latest/format.html#notes) section and there also mention that this happens by calling either one or the other scipy functions with links to their docs.